ethereum/core/src/main/java/org/hyperledger/besu/ethereum/bonsai/BonsaiWorldStateKeyValueStorage.java

```diff
@@ -193,13 +226,27 @@ public void clearFlatDatabase() {
     storageStorage.clear();
   }
 
+  public void clearSnapshot(final long blockNumber) {
+    final int bucketNumber = (int) ((blockNumber / BUCKET_SIZE) % 2);
+    final int cleanBucketCounter = BUCKET_SIZE - (int) (blockNumber % BUCKET_SIZE);
+    if (cleanBucketCounter == 1) {
+      ((bucketNumber == 0)
+              ? snapshotTrieBranchStorage.getSecond()
+              : snapshotTrieBranchStorage.getFirst())
+          .clear();
+    }
+  }
+
```

Comment:
If I am reading correctly, we wont be writing to the next bucket until we reach the 128 block boundary, and the current bucket gets wiped at that boundary.  this will leave us with empty buckets at 128 block boundaries won't it?

Reply:
indeed I was wrong. we need buckets of 256 to be ok

Reply:
or we could keep it at 128 and write to both buckets simultaneously, and keep the clearing behavior as is.

Reply:
You would still have to calculate which bucket to read from, but it would also remove the requirement to switch buckets on a cache miss.  Doubling write load to cut read load in half.  Depends on which is done more often.